tufrepo/cli.py

```diff
@@ -288,18 +278,42 @@ def set_expiry(ctx: Context, expiry: Tuple[int, str]):
 
 @edit.command()
 @click.pass_context
-@click.argument("delegate")
-def add_key(ctx: Context, delegate: str):
-    """Add new signing key for delegated role DELEGATE
+@click.option("--delegate")
+def add_key(ctx: Context, delegate: Optional[str] = None):
+    """Add a new signing key for a delegated role or succinct hash bin
+    delegations.
+
+    If a user wants to add a key to a single DELEGATE role then the role name
+    must be provided with the "--delegate" option.
+
+    If "--delegate" option is not provided tufrepo will assume you want to add
+    the new key to succinct hash bin delegation.
 
     The private key secret will be written to privkeys.json."""
     delegator = ctx.obj.role
     keyring: InsecureFileKeyring = ctx.obj.keyring
     key = keyring.generate_key()
 
-    with ctx.obj.repo.edit(delegator) as signed:
-        helpers.add_key(signed, delegator, delegate, key.public)
-    keyring.store_key(delegate, key)
+    targets: Targets
+    with ctx.obj.repo.edit(delegator) as targets:
+
+        # Add a key to one standard role.
+        if delegate is not None:
+            helpers.add_key(targets, delegator, delegate, key.public)
+            keyring.store_key(delegate, key)
+
+        # Add a key to succinct hash bin delegations.
+        else:
+            if targets.delegations is None or targets.delegations.succinct_roles is None:
+                raise ClickException(
+                    "'ROLE' doesn't contain information about succinct delegations"
+                )
+
+            helpers.add_key(targets, delegator, None, key.public)
+
+            # Store the generated key in each of the bins.
```

Comment:
nitpick: we do not store the key in the bins. We insert the key into the keyring once for each bin (at least until keyring can be smarter about this).